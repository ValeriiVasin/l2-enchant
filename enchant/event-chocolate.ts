import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const eventChocolateEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.Chocolate,
  chances: [65, 45, 40, 35, 30, 45, 38, 33, 28, 20, 28, 18, 12, 7],
  items: [
    Item.ChocolateFigure_1,
    Item.ChocolateFigure_2,
    Item.ChocolateFigure_3,
    Item.ChocolateFigure_4,
    Item.ChocolateFigure_5,
    Item.ChocolateFigure_6,
    Item.ChocolateFigure_7,
    Item.ChocolateFigure_8,
    Item.ChocolateFigure_9,
    Item.ChocolateFigure_10,
    Item.ChocolateFigure_11,
    Item.ChocolateFigure_12,
    Item.ChocolateFigure_13,
    Item.ChocolateFigure_14,
    Item.ChocolateFigure_15,
  ],
  boundaries: new Set([Item.ChocolateFigure_6, Item.ChocolateFigure_11]),
});
