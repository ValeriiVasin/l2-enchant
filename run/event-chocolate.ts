import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.ChocolateFigure_1, itemsRange(Item.ChocolateFigure_2, Item.ChocolateFigure_15), 100_000);
