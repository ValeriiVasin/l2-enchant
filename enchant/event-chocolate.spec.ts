import { eventChocolateEnchantMap } from './event-chocolate';

it('enchant map: event chocolate', () => {
  expect(eventChocolateEnchantMap).toMatchSnapshot();
});
