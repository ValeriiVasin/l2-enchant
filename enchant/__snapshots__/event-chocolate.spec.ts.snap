// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: event chocolate 1`] = `
Map {
  "Шоколадная Фигурка Ур.1" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.1",
    "results": Array [
      Object {
        "chance": 65,
        "item": "Шоколадная Фигурка Ур.2",
      },
      Object {
        "chance": 35,
        "item": "Шоколадная Фигурка Ур.1",
      },
    ],
  },
  "Шоколадная Фигурка Ур.2" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.2",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Шоколадная Фигурка Ур.3",
      },
      Object {
        "chance": 55,
        "item": "Шоколадная Фигурка Ур.1",
      },
    ],
  },
  "Шоколадная Фигурка Ур.3" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.3",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Шоколадная Фигурка Ур.4",
      },
      Object {
        "chance": 60,
        "item": "Шоколадная Фигурка Ур.1",
      },
    ],
  },
  "Шоколадная Фигурка Ур.4" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.4",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Шоколадная Фигурка Ур.5",
      },
      Object {
        "chance": 65,
        "item": "Шоколадная Фигурка Ур.1",
      },
    ],
  },
  "Шоколадная Фигурка Ур.5" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.5",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Шоколадная Фигурка Ур.6",
      },
      Object {
        "chance": 70,
        "item": "Шоколадная Фигурка Ур.1",
      },
    ],
  },
  "Шоколадная Фигурка Ур.6" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.6",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Шоколадная Фигурка Ур.7",
      },
      Object {
        "chance": 55,
        "item": "Шоколадная Фигурка Ур.6",
      },
    ],
  },
  "Шоколадная Фигурка Ур.7" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.7",
    "results": Array [
      Object {
        "chance": 38,
        "item": "Шоколадная Фигурка Ур.8",
      },
      Object {
        "chance": 62,
        "item": "Шоколадная Фигурка Ур.6",
      },
    ],
  },
  "Шоколадная Фигурка Ур.8" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.8",
    "results": Array [
      Object {
        "chance": 33,
        "item": "Шоколадная Фигурка Ур.9",
      },
      Object {
        "chance": 67,
        "item": "Шоколадная Фигурка Ур.6",
      },
    ],
  },
  "Шоколадная Фигурка Ур.9" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.9",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Шоколадная Фигурка Ур.10",
      },
      Object {
        "chance": 72,
        "item": "Шоколадная Фигурка Ур.6",
      },
    ],
  },
  "Шоколадная Фигурка Ур.10" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.10",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Шоколадная Фигурка Ур.11",
      },
      Object {
        "chance": 80,
        "item": "Шоколадная Фигурка Ур.6",
      },
    ],
  },
  "Шоколадная Фигурка Ур.11" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.11",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Шоколадная Фигурка Ур.12",
      },
      Object {
        "chance": 72,
        "item": "Шоколадная Фигурка Ур.11",
      },
    ],
  },
  "Шоколадная Фигурка Ур.12" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.12",
    "results": Array [
      Object {
        "chance": 18,
        "item": "Шоколадная Фигурка Ур.13",
      },
      Object {
        "chance": 82,
        "item": "Шоколадная Фигурка Ур.11",
      },
    ],
  },
  "Шоколадная Фигурка Ур.13" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.13",
    "results": Array [
      Object {
        "chance": 12,
        "item": "Шоколадная Фигурка Ур.14",
      },
      Object {
        "chance": 88,
        "item": "Шоколадная Фигурка Ур.11",
      },
    ],
  },
  "Шоколадная Фигурка Ур.14" => Object {
    "cost": "Шоколад",
    "item": "Шоколадная Фигурка Ур.14",
    "results": Array [
      Object {
        "chance": 7,
        "item": "Шоколадная Фигурка Ур.15",
      },
      Object {
        "chance": 93,
        "item": "Шоколадная Фигурка Ур.11",
      },
    ],
  },
}
`;
